### ISSUE TITLE
Help proxy for R errors

### ISSUE THREAD

**Issue by iamYannC** [posit-dev/positron#3753]
## Positron Version:
Positron Version: 2024.06.1 (system setup) build 2024.06.1-27
Code - OSS Version: 1.90.0
Commit: a893e5b282612ccb2200102957ac38d3c14e5196
Date: 2024-06-26T01:33:58.809Z
Electron: 29.4.0
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Windows_NT x64 10.0.22631
## Steps to reproduce the issue:

1. type `?data_frame` or any other `?`
2.

## What did you expect to happen?
Open a help window

## Were there any error messages in the output or Developer Tools console?

No error msg, but it just shows the html address and does not display the help documentation (only when hover, which is nice but not enough)
![image](https://github.com/posit-dev/positron/assets/130578277/4ef10a46-41f6-4b75-96d5-aa12ab02fb50)

**Comment by juliasilge** [comment]
Seems like this is likely a problem with Windows only, as I cannot reproduce this on a Mac. Can you go to "Output" and see if there are related errors, such as in the "Extension Host" output channel?

**Comment by GitHunter0** [comment]
`?data.frame` is working fine for me in Windows 10 with Positron 2024-06.1-27.

By the way, why `F1` shortcut to open **Help** only works for .py files? It is an important time saving feature, it would be great to work for .R files too (as in RStudio).

**Comment by DavisVaughan** [comment]
@GitHunter0 it does work for R files

**Comment by GitHunter0** [comment]
> @GitHunter0 it does work for R files

Sorry @DavisVaughan, in an .R file it is indeed working (my previous test was in a Quarto .qmd file).

**Comment by iamYannC** [comment]
> Seems like this is likely a problem with Windows only, as I cannot reproduce this on a Mac. Can you go to "Output" and see if there are related errors, such as in the "Extension Host" output channel?

Hey. Still with this bug.
I am not sure what you meant by Ext host, but i did find the log file, it's a bit long but here are the first lines, if that help a bit
```
2024-07-01 19:05:14.340 [info] Extension host with pid 16168 started
2024-07-01 19:05:14.340 [info] Skipping acquiring lock for c:\Users\97253\AppData\Roaming\Positron\User\workspaceStorage\5c969ba0f57a1e99e1f000f15ab1f060.
2024-07-01 19:05:14.358 [info] ExtensionService#_doActivateExtension vscode.emmet, startup: false, activationEvent: 'onLanguage'
2024-07-01 19:05:14.372 [info] ExtensionService#_doActivateExtension vscode.jupyter-adapter, startup: false, activationEvent: 'onLanguage:r', root cause: vscode.positron-r
2024-07-01 19:05:14.408 [info] ExtensionService#_doActivateExtension vscode.git-base, startup: false, activationEvent: '*', root cause: vscode.git
2024-07-01 19:05:14.434 [info] ExtensionService#_doActivateExtension vscode.positron-r, startup: false, activationEvent: 'onLanguage:r'
2024-07-01 19:05:14.464 [info] ExtensionService#_doActivateExtension vscode.git, startup: false, activationEvent: '*'
2024-07-01 19:05:14.509 [info] ExtensionService#_doActivateExtension vscode.github, startup: false, activationEvent: '*'
```
Otherwise, In the OUTPUT tab (next to problems, ports, debug etc..) there is not input.
![image](https://github.com/posit-dev/positron/assets/130578277/6c736786-eca4-4d49-8a97-05649c24779d)

--
F1: I get this pop-up:
![image](https://github.com/posit-dev/positron/assets/130578277/c0f891fe-592f-46f3-ad7d-a24628c969f4)

**Comment by juliasilge** [comment]
Can you run the command "Developer: Toggle Developer Tools" and take a look for logs related to Help? For me, I see this;

![dev-tools-help](https://github.com/posit-dev/positron/assets/12505835/61fb9ae0-18c2-43ab-85af-8f4924806792)

Which includes:

>  INFO PositronHelpService language runtime server sent show help event for: http://127.0.0.1:53548/library/base/html/data.frame.html

**Comment by iamYannC** [comment]
> PositronHelpService langu

Filtering for `help`
![image](https://github.com/posit-dev/positron/assets/130578277/07fe7cb7-98ac-443b-8e3d-7a19f29fe195)
[vscode-app-1719906795045.log](https://github.com/user-attachments/files/16064193/vscode-app-1719906795045.log)

Filtering for `data.frame` (the example i used)
![image](https://github.com/posit-dev/positron/assets/130578277/18b662f5-5b32-418f-ac15-8c6cd4921244)
[vscode-app-1719906915661.log](https://github.com/user-attachments/files/16064223/vscode-app-1719906915661.log)

And there was also that log (might be temp address? it changed):
[vscode-app-1719906634973.log](https://github.com/user-attachments/files/16064246/vscode-app-1719906634973.log)

Let me know if there is any additional info needed for you guys :)

**Comment by iamYannC** [comment]
Ok. I played with it with screen sharing and at one point *something* worked, but far less responsive than in rstudio..
When a small note appears in the bottom right its when i pressed F1 (with Fn button active)

https://github.com/posit-dev/positron/assets/130578277/2eb6f926-0974-499f-98dc-765b9907d59d

**Comment by nstrayer** [comment]
Is it possible to try this with python? E.g. switch to a python interpreter and run a help-generating command like `?load`?

**Comment by iamYannC** [comment]
> @iamYannC, If you have the time, could you try this with python? E.g. switch to a python interpreter and run a help-generating command like `?load`?

Python's help works. takes a sec to load, but help page is loaded. tried with `?load`, `?range` and `?%copy' and 'pd.read_csv' - all works fine

**Comment by nstrayer** [comment]
Okay cool -- thanks for checking! This means something is going wrong in the R language pack end. @DavisVaughan or @lionel- does that seem like a correct assessment?

**Comment by jennybc** [comment]
Observation re: the 3rd log ([vscode-app-1719906634973.log](https://github.com/user-attachments/files/16064246/vscode-app-1719906634973.log)):

I see the proxy being created at port 51197:

```
workbench.desktop.main.js:61147 [Extension Host] [HPM] Proxy created: /  -> http://127.0.0.1:51197
```

But the 502 error looks like a request on port 51301:

```
workbench.desktop.main.js:127455 DEBUG Webview(e2477533-89ab-4388-9864-96255b8e2083): webview ready
127.0.0.1:51301/library/base/html/data.frame.html:1 
        
        
       Failed to load resource: the server responded with a status of 502 (Bad Gateway)
```

In equivalent logging in my own Positron session just now, these two ports are the same (61588, yes, I know these ports are random):

```
[Extension Host] [HPM] Proxy created: /  -> http://127.0.0.1:61588
...
INFO PositronHelpService language runtime server sent show help event for: http://127.0.0.1:61588/library/base/html/data.frame.html
```

**Comment by iamYannC** [comment]
> Observation re: the 3rd log ([vscode-app-1719906634973.log](https://github.com/user-attachments/files/16064246/vscode-app-1719906634973.log)):
> 
> I see the proxy being created at port 51197:
> 
> ```
> workbench.desktop.main.js:61147 [Extension Host] [HPM] Proxy created: /  -> http://127.0.0.1:51197
> ```
> 
> But the 502 error looks like a request on port 51301:
> 
> ```
> workbench.desktop.main.js:127455 DEBUG Webview(e2477533-89ab-4388-9864-96255b8e2083): webview ready
> 127.0.0.1:51301/library/base/html/data.frame.html:1 
>         
>         
>        Failed to load resource: the server responded with a status of 502 (Bad Gateway)
> ```
> 
> In equivalent logging in my own Positron session just now, these two ports are the same (61588, yes, I know these ports are random):
> 
> ```
> [Extension Host] [HPM] Proxy created: /  -> http://127.0.0.1:61588
> ...
> INFO PositronHelpService language runtime server sent show help event for: http://127.0.0.1:61588/library/base/html/data.frame.html
> ```

This is wayy too much SE for me but If needed, I can go on a remote session or if you guide me like @juliasilge did, i hope I could provide some other usefull logs.

I did not try from other computers so can't say if its something in my local machine or in-fact some bug.

Thanks for the attention!

**Comment by juliasilge** [comment]
@iamYannC can you:

- download the latest release: https://github.com/posit-dev/positron/releases
- open up Positron and toggle Developer Tools on
- request help, such as via `?data.frame`
- look for two elements in the logs with ports

One will look like this:

> [Extension Host] [HPM] Proxy created: /  -> http://127.0.0.1:59984 

The other will look like this:

>  INFO PositronHelpService language runtime server sent show help event for: http://127.0.0.1:59984/library/stats/html/lm.html

Notice that I, like @jennybc, have the _same_ port in these two places. Do you have different ports?

**Comment by iamYannC** [comment]
> @iamYannC can you:
> 
> * download the latest release: https://github.com/posit-dev/positron/releases
> * open up Positron and toggle Developer Tools on
> * request help, such as via `?data.frame`
> * look for two elements in the logs with ports
> 
> One will look like this:
> 
> > [Extension Host] [HPM] Proxy created: /  -> http://127.0.0.1:59984
> 
> The other will look like this:
> 
> > INFO PositronHelpService language runtime server sent show help event for: http://127.0.0.1:59984/library/stats/html/lm.html
> 
> Notice that I, like @jennybc, have the _same_ port in these two places. Do you have different ports?

Hey. I cant understand its behaviour. It doesnt work, but then I just load a package (here, dplyr), and it's as if it brings up also the base R help documentation.
I did another video. I hope it also gives a clue as to what might happen. I tried either using `?` or with pressing `F1`

--
![image](https://github.com/posit-dev/positron/assets/130578277/70fca03b-558e-4b66-8164-f3afec313693)

3 Log files for data frames
[vscode-app-1720035176864.log](https://github.com/user-attachments/files/16089999/vscode-app-1720035176864.log)
[vscode-app-1720035192394.log](https://github.com/user-attachments/files/16090000/vscode-app-1720035192394.log)
[vscode-app-1720035200716.log](https://github.com/user-attachments/files/16090001/vscode-app-1720035200716.log)

![image](https://github.com/posit-dev/positron/assets/130578277/e713ee21-2df6-4a0f-8da2-45a28227d57d)
2 Log files
[vscode-app-1720035326995.log](https://github.com/user-attachments/files/16090019/vscode-app-1720035326995.log)
[vscode-app-1720035337750.log](https://github.com/user-attachments/files/16090020/vscode-app-1720035337750.log)

--
Here is what I did:

https://github.com/posit-dev/positron/assets/130578277/6484fb29-149f-4738-8f58-442090a72575

**Comment by juliasilge** [comment]
@iamYannC do you mind rebooting your computer and then downloading the [latest release](https://github.com/posit-dev/positron/releases) to try again?

**Comment by iamYannC** [comment]
> @iamYannC do you mind rebooting your computer and then downloading the [latest release](https://github.com/posit-dev/positron/releases) to try again?
```
Positron Version: 2024.07.0 (system setup) build 2024.07.0-17
Code - OSS Version: 1.90.0
Commit: 203958161106d234eaa85556293c1be8f8b395b3
Date: 2024-07-08T03:22:52.213Z
Electron: 29.4.0
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Windows_NT x64 10.0.22631
```
Same behviour.
Trying `?data.frame` or `?data.frame()` doesn't work. but then I just call a random help on a user-library function (without even loading, e.g `?glue::glue`) and after that, `?data.frame` works
----
**Failed to load resource: the server responded with a status of 502 (Bad Gateway)** - data.frame.html
[vscode-app-1720475758729.log](https://github.com/user-attachments/files/16134776/vscode-app-1720475758729.log)
--
On my second try it didnt load `glue` for the first time, so i re-ran it and it worked again..

https://github.com/posit-dev/positron/assets/130578277/c871f5ee-6e8e-483c-b919-c8df549e3407
Logs for failing `??glue` and `?glue::glue`
[vscode-app-1720476182716 - log for calling glue package.txt](https://github.com/user-attachments/files/16134821/vscode-app-1720476182716.-.log.for.calling.glue.package.txt)
[vscode-app-1720476212101 - log for calling glue function [failure].txt](https://github.com/user-attachments/files/16134822/vscode-app-1720476212101.-.log.for.calling.glue.function.failure.txt)

**Comment by DavisVaughan** [comment]
It would also be helpful to see your `Console: R x.y.z` output channel logs.

- Toggle Command Palette
- Search for `Output: Show output channels`
- Choose the one for `Console: R x.y.z` where `x.y.z` is your version
- Copy the entire log and paste it in or upload as a `.log` file

<img width="780" alt="Screenshot 2024-07-09 at 6 00 02 PM" src="https://github.com/posit-dev/positron/assets/19150088/eec03f35-a596-4a93-b3cb-b77cf7643013">

**Comment by iamYannC** [comment]
https://vscode.dev/github/iamYannC/r-podcast/blob/mainron/logs/20240709T214903/window1/exthost/vscode.jupyter-adapter/Console%20R%204.4.0.log#L8
[Console R 4.4.0.log](https://github.com/user-attachments/files/16148002/Console.R.4.4.0.log)
[Console R 4.4.0.log](https://github.com/user-attachments/files/16148003/Console.R.4.4.0.log)

I think i keep getting these SEND heartbeat w tout 30 secs line printed..
I am curious to know if i am the only one with these issues

**Comment by DavisVaughan** [comment]
Did you try and perform a help request before you captured those logs? I don't see a help request attempt anywhere in the logs. The heartbeat is normal.

**Comment by dengyuyueyuyue** [comment]
I also have the same issue: when typing ??function or help, the help window does not display properly and only shows a web link. 
![image](https://github.com/user-attachments/assets/285d67bd-f03f-4404-8462-1042ea85ffad)

**Comment by iamYannC** [comment]
> Did you try and perform a help request before you captured those logs? I don't see a help request attempt anywhere in the logs. The heartbeat is normal.

[Console R 4.4.0.log](https://github.com/user-attachments/files/16203563/Console.R.4.4.0.log)

**Comment by iamYannC** [comment]
> I also have the same issue: when typing ??function or help, the help window does not display properly and only shows a web link. ![image](https://private-user-images.githubusercontent.com/85548361/348470532-285d67bd-f03f-4404-8462-1042ea85ffad.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEwNDQ3MDIsIm5iZiI6MTcyMTA0NDQwMiwicGF0aCI6Ii84NTU0ODM2MS8zNDg0NzA1MzItMjg1ZDY3YmQtZjAzZi00NDA0LTg0NjItMTA0MmVhODVmZmFkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE1VDExNTMyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE0Yjc0NjFiYzE4YzIwYzRjZTU5ZjIyM2QzYjE2ZWYxNTZjYjBkMTQyZTY3ZjE5OTFmNTgwOWNiYmE4NWJmNjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-U0XwYGEZw0mIN1bpyFK0fa3zrZGpop-5cmn-vMzmcI)

https://youtu.be/DfxWhM8YTcw?t=202
Seems like we are not alone, also in python

**Comment by juliasilge** [comment]
I'm bumping this back to triage since multiple people are running into this problem.

**Comment by DavisVaughan** [comment]
@iamYannC the `Console: R` logs are quite helpful, thanks!

**Comment by DavisVaughan** [comment]
@iamYannC okay one more request. Please do exactly what you did before with the logs, but first go into Settings (`Cmd + ,` on Mac) and change this setting to `trace` and then completely restart Positron. Forgot to tell you that earlier, so sorry.

<img width="524" alt="Screenshot 2024-07-15 at 6 13 35 PM" src="https://github.com/user-attachments/assets/556b3001-fe0c-43be-9792-153250a7a236">

**Comment by iamYannC** [comment]
> @iamYannC okay one more request. Please do exactly what you did before with the logs, but first go into Settings (`Cmd + ,` on Mac) and change this setting to `trace` and then completely restart Positron. Forgot to tell you that earlier, so sorry.
> 
> <img alt="Screenshot 2024-07-15 at 6 13 35 PM" width="524" src="https://private-user-images.githubusercontent.com/19150088/348868717-556b3001-fe0c-43be-9792-153250a7a236.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE1NjMzODQsIm5iZiI6MTcyMTU2MzA4NCwicGF0aCI6Ii8xOTE1MDA4OC8zNDg4Njg3MTctNTU2YjMwMDEtZmUwYy00M2JlLTk3OTItMTUzMjUwYTdhMjM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIxVDExNTgwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAwNjk2NDllNzc3NjdjNjEyZjk0NTdhMjYyZTdlMDZhZWQ4ZDA3ODEwZDY5NDU4OTgxNWVhMTJjMTdlNTJjN2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.S9TI9Uyl6wjmkH_i5X1vHVy5uxpONOLYXgwelyCsQLs">

[Console R 4.4.0 with trace.log](https://github.com/user-attachments/files/16324297/Console.R.4.4.0.with.trace.log)
I restarted positron after modifying the R log, but I still got this "Debug":
![image](https://github.com/user-attachments/assets/8559a416-1715-40cb-b9d0-49c255716207)

Just in case, I will restart and do that `?mean` one more time, this time with a `trace` also in the console window:
[Console R 4.4.0 with trace 2.log](https://github.com/user-attachments/files/16324332/Console.R.4.4.0.with.trace.2.log)

**Comment by CongSIBR** [comment]
I have the same problem, which never happened before. Even after I restart my Mac and updated Positron.

Positron Version: 2024.07.0 (Universal) build 85
Code - OSS Version: 1.91.0
Commit: 9dfa34e1eb5a42ebbd1c1d17b55bef7e7bf0bf1f
Date: 2024-07-22T04:48:59.704Z
Electron: 29.4.0
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Darwin arm64 23.5.0


[2024-07-22_positron.log](https://github.com/user-attachments/files/16330306/2024-07-22_positron.log)
<img width="1728" alt="Screenshot 2024-07-22 at 16 32 18" src="https://github.com/user-attachments/assets/47dae327-f3d0-4201-822c-a6cd9a7d23e4">

**Comment by CongSIBR** [comment]
It's amazing, after working with VS Code for a while, I accidentally discovered that the positron issue was resolved! It's working now! **I can see the help information.** 
I wonder if this issue could be related to VS Code.

**Comment by juliasilge** [comment]
We haven't yet linked this discussion where folks are having similar problems: https://github.com/posit-dev/positron/discussions/3738

**Comment by DavisVaughan** [comment]
A couple thing I see in @iamYannC's latest logs

```
[R]   2024-07-21T12:06:27.854158Z  INFO  Proxying URL '"http://localhost:22825/library/base/html/mean.html"' path '/library/base/html/mean.html' content-type is 'Some("text/html")'
[R]     at crates\ark\src\help_proxy.rs:141
[R] 
[Positron] SEND heartbeat with timeout of 30 seconds
[R]   2024-07-21T12:06:32.224585Z  INFO  Proxying URL '"http://localhost:22825/doc/html/prism.css"' path '/doc/html/prism.css' content-type is 'Some("text/css")'
[R]     at crates\ark\src\help_proxy.rs:141
[R] 
[R]   2024-07-21T12:06:32.432514Z  INFO  Proxying URL '"http://localhost:22825/doc/html/prism.js"' path '/doc/html/prism.js' content-type is 'Some("text/javascript")'
[R]     at crates\ark\src\help_proxy.rs:141
[R] 
[R]   2024-07-21T12:06:32.638168Z  INFO  Proxying URL '"http://localhost:22825/doc/html/R.css"' path '/doc/html/R.css' content-type is 'Some("text/css")'
[R]     at crates\ark\src\help_proxy.rs:141
[R] 
[R]   2024-07-21T12:06:32.851731Z  INFO  Proxying URL '"http://localhost:22825/doc/html/katex-config.js"' path '/doc/html/katex-config.js' content-type is 'Some("text/javascript")'
[R]     at crates\ark\src\help_proxy.rs:141
[R] 
[R]   2024-07-21T12:06:33.060498Z  INFO  Proxying URL '"http://localhost:22825/doc/html/katex/katex.css"' path '/doc/html/katex/katex.css' content-type is 'Some("text/css")'
[R]     at crates\ark\src\help_proxy.rs:141
[R] 
[R]   2024-07-21T12:06:33.181138Z ERROR  Error proxying http://localhost:22825/doc/html/katex/katex.js: error sending request for url (http://localhost:22825/doc/html/katex/katex.js): operation was canceled: connection closed before message completed
[R]     at crates\ark\src\help_proxy.rs:177
```

And then a little later after a restart of R and another `?mean` call

```
[R]   2024-07-21T12:07:05.690377Z ERROR  Error proxying http://localhost:18876/library/base/html/mean.html: error sending request for url (http://localhost:18876/library/base/html/mean.html): connection error: An existing connection was forcibly closed by the remote host. (os error 10054)
[R]     at crates\ark\src\help_proxy.rs:177
```

**Comment by DavisVaughan** [comment]
Also see this in @CongSIBR's logs

```
[R]   2024-07-22T08:32:01.448273Z ERROR  Error proxying http://localhost:28336/library/base/html/cut.html: error sending request for url (http://localhost:28336/library/base/html/cut.html): connection closed before message completed
[R]     at crates/ark/src/help_proxy.rs:177
```

**Comment by DavisVaughan** [comment]
After a bit of reading, it is possible that this error message is a known thing with users of reqwest/hyper
https://github.com/hyperium/hyper/issues/2136

reqwest doesn't seem to have a retry mechanism baked in, which kind of blows my mind, but possibly we can try and add one using the again crate and this blog post as a model
https://stonecharioteer.com/posts/2022/rust-reqwest-retry.html

Or reqwest-retry, whichever seems more robust
https://docs.rs/reqwest-retry/latest/reqwest_retry/

That's a first step, at least

**Comment by iamYannC** [comment]
Very much appreciated @Posit team.
I will keep getting newest releases, hopefully it works out. If so I will post here

**Comment by CongSIBR** [comment]
Hi,
I’d like to know if this issue has been truly resolved. Since I encountered this problem last time, I’ve updated POSITRON two more versions, but the help information still doesn’t work, and the log information is still the same as before. Thank you.

**Comment by kv9898** [comment]
For me the ? command works but the ?? command doesn't. The problem can be reproduced on Windows and MacOS (see #4320). Is there a temporary workaround?

**Comment by CorradoLanera** [comment]
I can report a similar strange behavior: when asking for help in the console, a separate window (the classical R help windows) will be opened in the background, while if I press F1 on the function name, the positron help page will be opened (as expected and desired).

Is this related to this issue, or should I set some configuration for consistent behavior?

https://github.com/user-attachments/assets/be3c0519-d9d9-4c27-95f4-0e148c5e914b

Positron Version: 2024.08.0 (system setup) build 48
Code - OSS Version: 1.91.0
Commit: ed616b36655fb24d116108bdd833f1321704315b
Date: 2024-08-19T03:30:12.447Z
Electron: 29.4.0
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Windows_NT x64 10.0.26120

**Comment by juliasilge** [comment]
That definitely looks related @CorradoLanera, yes. Thank you for the report!

**Comment by juliasilge** [comment]
## QA Notes

Note that no one internally has been able to observe this class of problems so far. 

In https://github.com/posit-dev/ark/pull/489 I added retries to the Help proxy request. 

- Once the ark version in Positron is bumped, we expect that all kinds of Help requests continue to work in our internal automated and manual testing, like `?lm` and `??lm`.
- Once a new public-facing build with this change integrated becomes available, we think (HOPE) that this will solve some or all of these problems observed with Help requests not showing up in the Help pane. We'll post here in this issue and in #3738 when such a build is available. 

> [!NOTE]  
> If you as a user continue to see problems (or see different kinds of problems), what is helpful is to [change the R kernel log level to trace](https://github.com/posit-dev/positron/issues/3753#issuecomment-2229528367) and [share the logs from the R Console](https://github.com/posit-dev/positron/issues/3753#issuecomment-2218086901) attached as a file to an issue. Let's have any users with further problems open a new issue because this one is getting hard to parse.

**Comment by testlabauto** [comment]
### Verified Fixed
    Positron Version(s) : 2024.08.0-83
    OS Version          : OSX & Windows

### Test scenario(s) 

Attempted to repro according to filing instructions with OSX and Windows and saw no issues.

Link(s) to TestRail test cases run or created:
N/A

**Comment by DavisVaughan** [comment]
@testlabauto FYI this was actually closed a bit prematurely, as https://github.com/posit-dev/positron/pull/4528 just pulled in the actual fix (it is unlikely you would have been able to trigger the bug anyways though, as none of us have ever actually seen it before)

**Comment by juliasilge** [comment]
For folks who have been having trouble with the R Help proxy, the [latest release](https://github.com/posit-dev/positron/releases/tag/2024.09.0-1) incorporates the changes we hope will help or solve the problem. Please give it a try!

> [!NOTE]  
> If you as a user continue to see problems (or see different kinds of problems), what is helpful is to [change the R kernel log level to trace](https://github.com/posit-dev/positron/issues/3753#issuecomment-2229528367) and [share the logs from the R Console](https://github.com/posit-dev/positron/issues/3753#issuecomment-2218086901) attached as a file to an issue. Let's have any users with further problems open a new issue. Thanks!
